src/runtime_src/core/tools/common/tests/TestBandwidthKernel.cpp

```diff
@@ -25,8 +36,241 @@ TestBandwidthKernel::run(std::shared_ptr<xrt_core::device> dev)
     ptree.put("status", test_token_failed);
     return ptree;
   }
-  std::string testcase = (name.find("vck5000") != std::string::npos) ? "versal_23_bandwidth.py" : "23_bandwidth.py";
-  runTestCase(dev, testcase, ptree);
-  
+  runTest(dev, ptree);
   return ptree;
 }
+
+static bool 
+is_emulation() {
+  bool ret = false;
+  char* xcl_mode = std::getenv("XCL_EMULATION_MODE");
+  if (xcl_mode != nullptr) {
+    ret = true;
+  }
+  return ret;
+}
+
+void
+TestBandwidthKernel::runTest(std::shared_ptr<xrt_core::device> dev, boost::property_tree::ptree& ptree)
+{
+  const auto bdf_tuple = xrt_core::device_query<xrt_core::query::pcie_bdf>(dev);
+  const std::string bdf = xrt_core::query::pcie_bdf::to_string(bdf_tuple);
+  const std::string test_path = findPlatformPath(dev, ptree);
+  const std::string b_file = findXclbinPath(dev, ptree); // bandwidth.xclbin
+  bool flag_s = false;
+
+  xrt::device device(dev->get_device_id());
+
+  if (test_path.empty()) {
+    logger(ptree, "Error", "Platform test path was not found.");
+    ptree.put("status", test_token_failed);
+    return;
+  }
+
+  auto retVal = validate_binary_file(b_file);
+  if (flag_s || retVal == EOPNOTSUPP) {
+    ptree.put("status", test_token_skipped);
+    return;
+  } else if (retVal != EXIT_SUCCESS) {
+    logger(ptree, "Error", "Unknown error validating bandwidth xclbin");
+    ptree.put("status", test_token_failed);
+    return;
+  }
```

Comment:
Thank you simplifying the logic here! The `validate_binary_file` function checks the stream like you said (returns EOPNOTSUPP if stream isn't good, EXIT_SUCCESS if it is); I think we use it in a few places, so that's why it was made a dedicated function. Should I go ahead and just check if the stream is good here?

Reply:
yeah, 
i think  just do
```
if (!b_file.good())
  //skip
```